src/Table/Table.tsx

```diff
@@ -0,0 +1,985 @@
+                    updatedSubrows = updatedSubrows.map((subRow: TableRow) => ({ ...subRow, selected: checked }));
+
+                    return { ...originalRow, selected: checked, subRows: updatedSubrows };
+                }
+            } else if (type === "subRow") {
+                if (originalRow.rowIndex === rowIndex) {
+                    updatedSubrows = originalRow.subRows.map((originalSubrow: TableRow) => {
+                        if (originalSubrow.rowIndex === selectedRow.rowIndex) {
+                            return { ...originalSubrow, selected: checked };
+                        }
+                        return originalSubrow;
+                    });
+                    return { ...originalRow, subRows: updatedSubrows, selected: false };
+                }
+            }
+            return originalRow;
+        });
+
+        return updatedRows;
+    };
+
+    const setDefaultTableRows = async () => {
+        const updatedRows: Array<TableRow> = await getRows(props.data);
+        setTableRows(updatedRows);
+        setTableRowsImage(updatedRows);
+    };
+
+    const doPaginate = (): void => {
+        if (props.currentpage && props.offset && (tableRows.length > 0)) {
+            // pagination start from 1 hence the need fro deducting 1
+            const start: number = (props.currentpage - 1) * props.offset;
+            const end: number = (props.offset * (props.currentpage));
+
+            const currentPage: Array<TableRow> = tableRows.slice(start, end);
+            setCurrentTableRows(currentPage);
+        } else {
+            setCurrentTableRows(tableRows);
+        }
+    };
+
+    const rowsAreCollapsable = (): boolean => {
+        return currentTableRows.some((row: TableRow) => {
+            return (
+                ((row.subRows.length > 0) || row.rowContentDetail) ||
+                (row.subRows.some((subRow: TableRow) => subRow.rowContentDetail || (row.subRows && row.subRows.length > 0)))
+            );
+        }) && !!props.onRowExpanded;
+    };
+
+    const doSearch = (): void => {
+        let searchResult: Array<TableRow> = [];
+        if (props.searchProps?.searchText && props.searchProps?.searchInColumns && props.searchProps?.searchInColumns.length > 0) {
+            searchResult = searchTextInArray(tableRowsImage, props.searchProps.searchText, props.searchProps.searchInColumns);
+        } else {
+            searchResult = [...tableRowsImage];
+        }
+
+        setTableRows(searchResult);
+        props.searchProps?.onSearch && props.searchProps?.onSearch(searchResult);
+    };
+
+    // useEffects ----------------------------------------------------------
+
+    // Adding event listener to listen to clicks outside the component on mount, removing on unmount
+    React.useEffect(() => {
+        document.addEventListener("mousedown", onClickOutside);
+        return () => {
+            document.removeEventListener("mousedown", onClickOutside);
+        };
+    });
```

Comment:
You need to pass an empty array to `useEffect` if you only need it to happen when it **mount** and **unmount**
```suggestion
    }, []);
```